File: project/project2/vectorf.cpp
#include <iostream>
#include <vector>
#include <ctime>
#include <chrono>

using namespace std;
using namespace chrono;

int main() {
    srand(time(NULL)); // 设置随机数种子

    int n;
    cout << "请输入向量的长度：";
    cin >> n;
    // 动态分配内存
    float* vec1 = new float[n]; 
    float* vec2 = new float[n];

    for (int i = 0; i < n; i++) {
        vec1[i] = static_cast<float>(rand()) / RAND_MAX; // 生成0-1之间的随机浮点数
    }

    for (int i = 0; i < n; i++) {
        vec2[i] = static_cast<float>(rand()) / RAND_MAX; // 生成0-1之间的随机浮点数
    }

    double result = 0.0;

    // 获取当前时间
    auto start_time = std::chrono::high_resolution_clock::now();

    for (int i = 0; i < n; i++){
        result += vec1[i] * vec2[i];
    }

    // 获取程序结束时间点
    auto end_time = std::chrono::high_resolution_clock::now();

    cout << "点乘结果为: " << result << endl;

    // 计算程序运行时间
    auto elapsed_ns = duration_cast<nanoseconds>(end_time - start_time);
    cout << "程序运行时间为：" << elapsed_ns.count() << "ns" << endl;

    delete[] vec1,vec2; // 释放动态分配的内存 
    
    return 0;
}

File: project/project2/vectorf.java
import java.util.Random;
import java.util.Scanner;

public class vectorf {
    public static void main(String[] args) {
        Random random = new Random(); // 创建一个随机数生成器
        
        System.out.print("请输入向量的长度：");
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        float[] vec1 = new float[n];
        float[] vec2 = new float[n];
        for (int i = 0; i < n; i++) {
            vec1[i] = random.nextFloat();
            vec2[i] = random.nextFloat();
        }

        double result = 0.0;
        long startTime = System.nanoTime(); // 记录程序开始执行的时间

        for (int i = 0; i < n; i++){
            result += vec1[i] * vec2[i];
        }

        long endTime = System.nanoTime(); // 记录程序结束执行的时间
        long duration = endTime - startTime; // 计算程序运行时间，单位为纳秒
        sc.close();
        System.out.println("点乘结果为：" + result);
        System.out.println("程序运行时间为：" + duration + " ns");

    }
}